File: open_days_backend/src/main/java/com/sapientia/open/days/backend/ui/model/request/EventModel.java
package com.sapientia.open.days.backend.ui.model.request;

import com.fasterxml.jackson.annotation.JsonProperty;

@SuppressWarnings("unused")
public class EventModel {

	private boolean isOnline;

	private String dateTime;
	private String meetingLink;
	private String organizerId;
	private String activityName;

	@JsonProperty("isOnline")
	public boolean getOnline() {
		return isOnline;
	}

	public String getDateTime() {
		return dateTime;
	}

	public String getMeetingLink() {
		return meetingLink;
	}

	public String getOrganizerId() {
		return organizerId;
	}

	public String getActivityName() {
		return activityName;
	}

	public void setOnline(boolean online) {
		isOnline = online;
	}

	public void setDateTime(String dateTime) {
		this.dateTime = dateTime;
	}

	public void setMeetingLink(String meetingLink) {
		this.meetingLink = meetingLink;
	}

	public void setOrganizerId(String organizerId) {
		this.organizerId = organizerId;
	}

	public void setActivityName(String activityName) {
		this.activityName = activityName;
	}
}

File: open_days_frontend/lib/models/event_model.dart
class EventModel {
  bool isOnline;
  String dateTime;
  String location;
  String? imagePath;
  String description;
  String organizerId;
  String activityName;
  String? meetingLink;

  EventModel({
    this.dateTime = '',
    this.location = '',
    this.isOnline = false,
    this.description = '',
    this.organizerId = '',
    this.activityName = '',
  });

  Map<String, dynamic> toJson() => {
        'isOnline': isOnline,
        'dateTime': dateTime,
        'location': location,
        'imageLink': imagePath,
        'description': description,
        'meetingLink': meetingLink,
        'organizerId': organizerId,
        'activityName': activityName,
      };
}
